rust-runtime/aws-smithy-runtime/tests/reconnect_on_transient_error.rs

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#![cfg(all(
+    feature = "client",
+    feature = "wire-mock",
+    feature = "connector-hyper-0-14-x",
+))]
+
+use ::aws_smithy_runtime::client::retries::classifier::{
+    HttpStatusCodeClassifier, SmithyErrorClassifier,
+};
+use ::aws_smithy_runtime_api::client::retries::RetryClassifiers;
+use aws_smithy_async::rt::sleep::TokioSleep;
```

Comment:
I'd expect the matching version of this test to be deleted?

Reply:
It will be in a future PR that more broadly cleans up middleware code.